schemas/groups/resources.json

```diff
@@ -220,6 +220,22 @@
               "type": "string",
               "description": "Location of the note"
             },
+            "group": {
+                "type": "string",
+                "description": "Name of a group or collection to which the note is attached"
+            }, 
+            "author": {
+                "type": "string",
+                "description": "Name of the creator of the note"
+            }, 
+            "readOnly": {
+                "type": "boolean",
+                "description": "Indicates whether the whether the content of the Note is permitted to be changed."
+            },
+            "published": {
+                "type": "boolean",
+                "description": "Indicates whether the note is available only for use on this server or shared with others."
```

Comment:
Can you explain this a bit more. The name of the property and the description don't quite connect for me. `Published` means that somebody published (emphasis on the past tense) this somehow somewhere. Do you mean that the note is `local=true` to this server?

Reply:
Same as above, better to control this withe per-key security. That gives fine-grain control over who can read/write the note.

Reply:
I believe the intent of this attribute is to support the use case where a note is being drafted (incomplete) and is saved to the server but not ready for publication / upload / sharing.
I think `local=true` works here.

Reply:
hmm..it might be better to have a `validity` attribute, eg from and to dates. That would allow publishing notes in the future and removing them, eg 'Notice to Mariners - logs in water around xxx' or weather warnings etc

Reply:
So something like ..
```
validity: {
    start: <datetime>,
    end: <datetime>
}
```
where:
- NO **start** and  NO **end**= note is valid from time of creation
- DEFINED **start** and NO **end**= note becomes valid after **start** date/time has elapsed and will remain valid indefinitely
- NO **start** and DEFINED **end**= note is valid only until **end** date/time has elapsed
- DEFINED **start** and DEFINED **end**= note is valid only between the **start**  and **end** date/times

Reply:
yes (`startTime` ,`endTime` for consistency maybe), and use it instead of `readOnly` and `published`. eg if its before `startTime` only the author can see/edit it. Use key-level security to control access when published. 

A background cron job can then auto publish/remove notes to suit.

Reply:
Unfortunately this doesn't support the 'save as Draft' functionality that `published` was intended to provide. The User doesn't want his note made valid until he physically goes in and marks it as so.

Reply:
The user could set a date in the future, even 09/09/2999 so the note would not be published. That could be triggered by a 'Draft/Publish' button ?

I'm not against the `published` key, just dont want unnecessary duplicated functionality, especially one-off for a specific key (`notes`). 
Implementing a server is complex enough now, the more we can use existing generic features the better.

Reply:
I take the point about duplication, and agree that a date range can be useful, but using a random future date for this is something of a distortion of the spec, since the functionality isn't implicit.
Also, the user will want to retrieve all 'draft' notes from the server, and this implies the client 'knowing' the magic date.  If the date is simply in the future, how does the client know the difference between 'draft' notes and 'finished' notes to be published at some point in the future?

Reply:
The inclusion of the `properties` block should cater for this.